imports/lib/schemas/customTypes.ts

```diff
@@ -38,7 +38,7 @@ export const allowedEmptyString = z.string();
 // always return values (e.g. createdTimestamp will always be a Date, it'll just
 // get set on insertion and not on subsequent updates).
 
-export const stringId = z.string().regex(Id).optional()
+export const stringId = nonEmptyString.optional()
   .transform((v) => v as unknown as string);
 attachCustomJsonSchema(stringId, { bsonType: 'string', pattern: Id.source }, true);
```

Comment:
Seems to me like this `attachCustomJsonSchema` call will still attempt to enforce that the id matches `Id` at the DB layer, which going by the commit message, is not what you were aiming for?